agent/structs/acl.go

```diff
@@ -113,8 +113,11 @@ type ACLIdentity interface {
 	SecretToken() string
 	PolicyIDs() []string
 	EmbeddedPolicy() *ACLPolicy
+	IsExpired(asOf time.Time) bool
 }
 
+var _ ACLIdentity = (*ACLToken)(nil)
```

Comment:
Why? Is this just to check that an ACLToken satisfies the interface? If so wouldn't that be caught with unit tests?

Reply:
Usually I prefer to pre-assert an interface like this when frequently modifying both a struct and an interface, especially when I'm not fully apprised of all of the ways the interface and struct are used throughout the code and tests. That way I know early on that I've forgotten a method or forgotten to update the signature during a refactor, rather than having it fail in stranger ways later.

This carried over from the larger branch where the ACLIdentity interface had more extensive changes. I totally overlooked this slipping into this PR.

Reply:
I guess it doesn't really hurt. You should add a comment about why this should be here then.